Samples/Swift/DaysUntilBirthday/Shared/ViewModels/VerifiedAgeViewModel.swift

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+import GoogleSignIn
+
+#if os(iOS)
+
+final class VerifiedAgeViewModel: ObservableObject {
+  /// The user's account verification status.
+  /// - note: This will publish updates when its value changes.
+  @Published var verificationState: VerificationState
+
+  private var loader: VerificationLoader {
+    return VerificationLoader(verifiedViewAgeModel: self)
+  }
+
+  /// Creates an instance of this view model.
+  init() {
+    self.verificationState = .unverified
+  }
+
+  /// Verifies the user.
+  func verifyAccountDetails() {
+    switch self.verificationState {
+    case .unverified:
+      loader.verifyAccountDetails()
```

Comment:
What is the role of this `switch`? I recognize that the user cannot literally change, but knowledge of the user's verified age can. I.e., perhaps that user disconnected the app from this scope. In this case, if the app is still in memory, then it would still reflect that the user is verified, but that is based on older knowledge.

Reply:
The `switch` is meant to prevent redundant verification checks so whenever an instance of `VerifiedAgeViewModel` is created then the user's state is set to `unverified` from the beginning. Though, I see the concern where the app should reflect the most up-to-date verification status when the user disconnected the app from the scope. In this case, then it would be helpful to add an additional check if the `verificationState` is labeled `verified`. If not, then we will force re-verification. Would it makes sense to make another request for the scopes?

Reply:
Though this also seems to apply for the sign-in flow. If I disconnect the app from the sign-in scopes, then the information is still there due to memory. Is this expected? Wondering if the verify flow should be making an extra request to verify that the scopes have changed or not. Is there a way to add a listener for this?

Reply:
There is no push driven listener we can implement for changes in sign in (or scopes) state changed in the user's account settings page. The issue I'm seeing is that the verified age signal is being treated as fixed once it's been fetched, but it isn't necessarily.

Reply:
Sounds good, changed it back by removing the switch and making the request whenever the view method is called.

Reply:
Discussed offline regarding refreshing the token whenever it's expired or near expiration. Since the refresh API is not included in this PR I'll add this in a separate PR after https://github.com/google/GoogleSignIn-iOS/pull/453.